nucypher/characters/lawful.py

```diff
@@ -281,40 +317,57 @@ def grant(self,
                     "know which nodes to use.  Either pass them here or when you make the Policy, "
                     "or run the learning loop on a network with enough Ursulas.".format(policy.n))
 
-        policy.make_arrangements(network_middleware=self.network_middleware,
-                                 handpicked_ursulas=handpicked_ursulas)
+        # Make policy arrangements with N Ursulas
+        policy.make_arrangements(network_middleware=self.network_middleware, handpicked_ursulas=handpicked_ursulas)
 
-        # REST call happens here, as does population of TreasureMap.
+        # On-chain publication, REST negotiation, and population of TreasureMap.
         policy.enact(network_middleware=self.network_middleware)
-        return policy  # Now with TreasureMap affixed!
+
+        # TODO: Add caching flags
+        self.add_active_policy(active_policy=policy)
+
+        # Create the policy's credential, and store it it Alice
+        # has a PolicyCredential Storage configured.
+        credential = policy.credential()
+        if self.credential_storage and self.store_policy_credentials:
+            self.credential_storage.save(credential=credential)
+
+        return policy
 
     def get_policy_encrypting_key_from_label(self, label: bytes) -> UmbralPublicKey:
         alice_delegating_power = self._crypto_power.power_ups(DelegatingPower)
         policy_pubkey = alice_delegating_power.get_pubkey_from_label(label)
         return policy_pubkey
 
-    def revoke(self, policy) -> Dict:
+    def revoke(self, policy) -> Union[Dict, Tuple[Dict, Dict]]:
         """
         Parses the treasure map and revokes arrangements in it.
         If any arrangements can't be revoked, then the node_id is added to a
         dict as a key, and the revocation and Ursula's response is added as
         a value.
         """
+
+        # TODO
+        if policy.treasure_map is not NO_DECRYPTION_PERFORMED:
```

Comment:
This line reads a little weird since it's a double negative. 

Wdyt about modifying the check and switching the order of the code blocks?

eg. 
```
if policy.treasure_map is NO_DECRYPTION_PERFORMED:
    revocation_kit = RevocationKit.from_policy(policy=policy, signer=self.stamp)
else:
    revocation_kit = RevocationKit.from_treasure_map(treasure_map=policy.treasure_map, signer=self.stamp)
```

Side note (personal preference): I would vote for the constant to be `NOT_DECRYPTED`.

Reply:
:heavy_check_mark: